tests/coerce.rs

```diff
@@ -0,0 +1,121 @@
+#![cfg(feature = "nightly-features")]
+
+use std::cell::RefCell;
+use shredder::*;
+
+type GcNode = Gc<RefCell<dyn LinkedListNode + 'static>>;
+
+struct LinkedListBuilder {
+    head: GcNode,
+    tail: Option<GcNode>,
+}
+
+impl LinkedListBuilder {
+    fn new(head: GcNode) -> Self {
+        let tail = head.get()
+            .borrow()
+            .tail();
+        Self { head, tail, }
+    }
+
+    fn next<T: LinkedListNode + 'static>(mut self, next: T) -> Self {
+        let next = Gc::new(RefCell::new(next));
+        let tail = match self.tail {
+            Some(tail) => tail.clone(),
+            None => self.head.clone(),
+        };
+        tail.get().borrow_mut().set_next(Some(next.clone()));
+        self.tail = Some(next);
+        self
+    }
+
+    fn finish(self) -> GcNode {
+        self.head
+    }
+}
+
+impl<T: LinkedListNode + 'static> From<T> for LinkedListBuilder {
+    fn from(other: T) -> Self {
+        Self::new(Gc::new(RefCell::new(other)))
+    }
+}
+
+trait LinkedListNode: Scan {
+    fn label(&self) -> &str;
+    fn next(&self) -> Option<GcNode>;
+    fn set_next(&mut self, next: Option<GcNode>);
+
+    fn tail(&self) -> Option<GcNode> {
+        match self.next() {
+            Some(next) => if next.get().borrow().next().is_some() {
+                next.get().borrow().tail()
+            } else {
+                self.next()
+            }
+            None => None,
+        }
+    }
+}
+
+macro_rules! make_node {
+    ($name:ident, $label:expr) => {
+
+        #[derive(Default, Debug, Scan)]
+        struct $name {
+            next: Option<GcNode>
+        }
+
+        impl LinkedListNode for $name {
+            fn label(&self) -> &str { $label }
+            fn next(&self) -> Option<GcNode> { self.next.clone() }
+            fn set_next(&mut self, next: Option<GcNode>) { self.next = next; }
+        }
+    };
+}
+
+make_node!(BlueNode, "blue");
+make_node!(RedNode, "red");
+make_node!(GreenNode, "green");
+make_node!(BlackNode, "black");
+
+#[test]
+fn coerce_nodes() {
```

Comment:
Note that if we ever have multiple tests in a single integration file, we need to serialize with a mutex. Not needed here since there is only one test

Reply:
Yup, I did have that guard in there previously, but I actually removed it since there was just that one test.